src/org/jitsi/impl/neomedia/codec/video/vp9/DePacketizer.java

```diff
@@ -166,5 +169,49 @@ public static int getSpatialLayerIndex(byte[] buf, int off, int len)
 
             return (buf[loff] & 0xE) >> 1;
         }
+        
+        /**
+         * Check if the current the packet contains a key frame
+         *
+         * @param buf the byte buffer that holds the VP9 payload.
+         * @param off the offset in the byte buffer where the VP9 payload starts.
+         * @param len the length of the VP9 payload.
+         * @return true if the frame is a key frame, false otherwise
+         */
+        public static boolean isKeyFrame(byte[] buf, int off, int len)
+        {
+            //This packet will have its P bit equal to zero, SID or D
+            //bit (described below) equal to zero, and B bit (described below)
+            //equal to 1
+            
+            if (!isValid(buf, off, len))
+            {
+                return false;
+            }
+
+            // P_BIT must be 0 and B_BIT 1
+            if((buf[off] & P_BIT) != 0 || (buf[off] & B_BIT) == 0)
```

Comment:
I think we need to check for the L_BIT as well.

Reply:
As far i understand we should check the L_BIT to ensure that SID and D can be checked below?
Is this correct?
if((buf[off] & P_BIT) != 0 || (buf[off] & B_BIT) == 0 ||  (buf[off] & L_BIT) == 0 )

Reply:
Yes 👍

Reply:
Actually, let me read through this a little bit more carefully.

Reply:
okay, I read it a bit more carefully, this is the correct condition, thanks!